File: camera-test/C++/camera-test.cpp
#include <opencv2/opencv.hpp>

using namespace cv;

int main() {
    VideoCapture cap(0);

    Mat input;
    bool pause = false;
    namedWindow("Camera");
    while (cap.isOpened()) {
        if (!pause) {
            cap >> input;
            imshow("Camera", input);
        }
        char c = (char) waitKey(1);
        if (c == 27) break;
        else if (c == ' ') pause = !pause;
    }
    cap.release();
    destroyAllWindows();
    return 0;
}

File: camera-test/python/camera-test.py
import cv2

cap = cv2.VideoCapture(0)
pause = False
cv2.namedWindow("Camera")
while cap.isOpened():
    if not pause:
        ret, frame = cap.read()
        cv2.imshow("Camera", frame)
        c = cv2.waitKey(1)
        if c == 27:
            break
        elif c & 0xFF == ord(' '):
            pause = not pause

cap.release()
cv2.destroyAllWindows()
